### PR TITLE
Implementation of Certificate Authority resource

### LINKED ISSUE #70: Add support for certificate authority.
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Add support for certificate authority resource and datasource

### Use Case(s)

Enable CA configuration from the terraform provider

### Potential Configuration

TBD

### References

NA

### PR DESCRIPTION
**Pull request by dimitarproynov**
Decided to go with a single resource with two "types" that represent the two distinct CA.

```bash
Testing done:
make build
make lint
make test
```

In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/terraform-provider-vcf/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

**Type of Pull Request**

- [ ] This is a bug fix.
- [X] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [x] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #70 

**Test and Documentation Coverage**

Will followup on tests in next commit

For bug fixes or features:

- [ ] Tests have been completed.
- [x] Documentation has been added/updated.

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

**Review by tenthirtyam — COMMENTED** [review]
One suggestion to perhaps add a country code validation. Perhaps by using a package like [`biter777/countries`](https://pkg.go.dev/github.com/biter777/countries) versus hard-coding. Another package might be preferable however.

**Review by vasilsatanasov — CHANGES_REQUESTED** [review]
As far as I remember there can be only one CA configured i.e. either Microsoft either SSL Would you confirm this and make a validation if so. Also IMHO it would be more customer friendly to have something like 
```
certificate_authority {
    microsoft {
       // microsoft properties
    }
    open_ssl {
      // openssl properties
    }
}
```
in the HCL since the properties are a lot and it is easy to mix up.

File: internal/provider/resource_certificate_authority.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+func ResourceCertificateAuthority() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCertificateAuthorityCreate,
+		ReadContext:   resourceCertificateAuthorityRead,
+		UpdateContext: resourceCertificateAuthorityUpdate,
+		DeleteContext: resourceCertificateAuthorityDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
```
Comment (raj-chelur) [inline internal/provider/resource_certificate_authority.go:34]: Long timeouts not required for these resources. They are very quick to create. You can make it same as the ceip resource above - 30 mins.

**Review by raj-chelur — APPROVED** [review]

File: internal/provider/resource_certificate_authority.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "OpenSSL CA domain name",
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"country": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "ISO 3166 country code where company is legally registered",
+							ValidateFunc: validation.StringInSlice([]string{"US", "CA", "AX", "AD", "AE", "AF", "AG", "AI", "AL", "AM", "AN", "AO", "AQ", "AR", "AS", "AT", "AU",
```
Comment (vasilsatanasov) [inline internal/provider/resource_certificate_authority.go:98]: Could you please extract the country codes array to some file where constants are kept.

**Review by vasilsatanasov — DISMISSED** [review]

**Review by tenthirtyam — APPROVED** [review]
LGTM!!

File: internal/provider/resource_certificate_authority.go (hunk 2 above)
Comment (dimitarproynov) [reply internal/provider/resource_certificate_authority.go:98]: Unfortunately in Go []string cannot be a constant, thus I've created GetIso3316CountryCodes() function in the constants package

**Review by vasilsatanasov — APPROVED** [review]
LGTM!

**Pull request merged**